### DISCUSSION TITLE
[Question]  Realtime muxer

### DISCUSSION THREAD

**Discussion by SonnyWalkman** [asticode/go-astits#60] [Q&A]
Hello @asticode,

great work here and I'm new to programming in Go! I've been using TSduck in exec to do what I need to however, a quick look at what you have done is promising.   How far off are implementation of TDT and TOT tables.?

I'm using Teletext PES and guessing the muxer is capable of muxing the PES from another program once formatted correctly. The teletext PES need its own PTS clock or does your mux do all the PCR and PTS metrics?

**Comment by asticode** [comment]
@SonnyWalkman thanks for your interest in the project 👍 

> How far off are implementation of TDT and TOT tables.?

`TOT` tables are demuxed (see [here](https://github.com/asticode/go-astits/blob/master/data_tot.go)) but not muxed and `TDT` tables are neither demuxed nor muxed.

> The teletext PES need its own PTS clock or does your mux do all the PCR and PTS metrics?

Right now only `PAT` and `PMT` tables can be generated manually using the [WriteTables](https://github.com/asticode/go-astits/blob/master/muxer.go#L297) method or automatically using the [MuxerOptTablesRetransmitPeriod](https://github.com/asticode/go-astits/blob/master/muxer.go#L65) option. The muxer's [WriteData](https://github.com/asticode/go-astits/blob/master/muxer.go#L165) method must be provided a `*MuxerData` which contains the information you want to write; `astits` doesn't "interfere" with this information therefore the teletext PES needs its own PTS.

**Comment by TVforME** [comment]
Thanks @asticode,

Is the other to do items going to happen anytime soon?
Its almost there however need NIT, TOT and TDT

**Comment by asticode** [comment]
I'll be honest, I'm not implementing new stuff myself in this repo anymore except for bug fixing, but I'm welcoming PRs which I can supervise and help on 👍

**Comment by TVforME** [comment]
Thanks @asticode for you reply,
I'll look to adding the features sometime in future. I've been able to do what I require with TSduck however a native GO muxer is my preferred option, however looking to developing  GO wrappers for TSduck since GO and C++  go (no pun intended) well together using cgo.  TSduck.go binding. TSduck has Python and Java binding why no Go?

**Reply by TVforME** [reply]
I've since managed to mux all of using combination of ffmpeg and pipe ffmpeg output into TSduck.
